scripts/typescript_checks.py

```diff
@@ -57,13 +812,107 @@ def validate_compiled_js_dir() -> None:
             'in %s: %s' % (COMPILED_JS_DIR, TSCONFIG_FILEPATH, out_dir))
 
 
+def compile_strict_tsconfig(
+    config_path: str, error_messages: List[str]
+) -> None:
+    """Compiles strict TS config with files those are neither strictly typed
+    nor present in TS_STRICT_EXCLUDE_PATHS. If there are any errors, we
+    restores the original config.
+
+    Args:
+        config_path: str. The config that should be used to run the typescript
+            checks.
+        error_messages: List[str]. A list of error messages produced by
+            compiling the strict typescript config.
+    """
+    # Generate file names from the error messages.
+    errors = [x.strip() for x in error_messages]
+    # Remove the empty lines and error explanation lines.
+    errors = [x for x in errors if x.startswith(PREFIXES)]
+    # Remove error explanation lines.
+    errors = [x.split('(', 1)[0] for x in errors]
+    # Remove the duplicate occurrences of the file names.
+    files_with_errors = sorted(set(errors))
+
+    # List of missing files that are neither strictly typed nor present in
+    # TS_STRICT_EXCLUDE_PATHS.
+    files_not_type_strict = []
+    for filename in files_with_errors:
+        if filename not in TS_STRICT_EXCLUDE_PATHS:
+            files_not_type_strict.append(filename)
+
+    # Add "typings" folder to get global imports while compiling.
+    files_not_type_strict.append('typings')
+
+    try:
+        # Update "include" field of tsconfig-strict.json with files those are
+        # neither strict typed nor present in TS_STRICT_EXCLUDE_PATHS.
+        # Example: List "files_not_type_strict".
+        with utils.open_file(STRICT_TSCONFIG_FILEPATH, 'r') as f:
+            strict_ts_config = yaml.safe_load(f)
+            strict_ts_config['include'] = files_not_type_strict
+
+        with utils.open_file(STRICT_TSCONFIG_FILEPATH, 'w') as f:
+            json.dump(strict_ts_config, f, indent=2, sort_keys=True)
+            f.write('\n')
+
+        # Compile tsconfig-strict.json with files those are neither strictly
+        # typed nor present in TS_STRICT_EXCLUDE_PATHS. All those files
+        # present inside include property.
+        os.environ['PATH'] = '%s/bin:' % common.NODE_PATH + os.environ['PATH']
+        validate_compiled_js_dir()
+
+        if os.path.exists(COMPILED_JS_DIR):
+            shutil.rmtree(COMPILED_JS_DIR)
+
+        cmd = ['./node_modules/typescript/bin/tsc', '--project', config_path]
+        process = subprocess.Popen(
```

Comment:
Hey, actually thinking about this, we can solve the problem with the file staying modified by instead naming the file differently. So you generate the file from tsconfig-strict.json and from the list of files, then you run the TS check and after that just remove the generated file. You can also add the generated file name to .gitignore so that it is never commited.

Reply:
Means one more separate file look like ts-config file and use that. Delete at last? then no worry about restoration and all.

Reply:
Yes

Reply:
Done, PTAL